### PR TITLE
Fw 5949 batch importing test coverage and edge cases

### PR DESCRIPTION
**Pull request by sahil97**
### Description of Changes
- Changed `alt_spelling` to `alternate_spelling`.
- Added cleaning (stripping, lowercasing, title-casing) to required attributes for DictionaryEntry resource.
- Added support for variations in input boolean values, e.g. True, yes, Y, 1 etc.
- Added tests for more edge cases.

### Checklist
- ~README / documentation has been updated if needed~
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~
- ~Insomnia workspace has been updated if applicable~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- ~Pull the branch and test locally~

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/resources/utils/import_export_widgets.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ class InvertedBooleanFieldWidget(Widget):
     """Import/export widget to return expected boolean value for audience related fields."""
 
     def clean(self, value, row=None, **kwargs):
-        curr_value = get_valid_boolean(value)
+        curr_value = get_valid_boolean_for_batch_import(value)
```
Comment (sarahfirstvoices) [inline firstvoices/backend/resources/utils/import_export_widgets.py:83]: I think this util function could be a method right in this widget, unless we are currently using it somewhere else

**Review by sarahfirstvoices — DISMISSED** [review]

**Review by sarahfirstvoices — APPROVED** [review]
